app/models/merchant.rb

```diff
@@ -9,19 +9,34 @@ class Merchant < ApplicationRecord
 
   def self.most_revenue(number = 5)
     select("merchants.*, sum(invoice_items.unit_price * invoice_items.quantity) as revenue_sum")
-            .joins(invoices: [:invoice_items, :transactions])
-            .group(:id)
-            .order("revenue_sum desc")
-            .where("transactions.result = ?", "success")
-            .limit(number)
+      .joins(invoices: [:invoice_items, :transactions])
+      .group(:id)
+      .order("revenue_sum desc")
+      .where("transactions.result = ?", "success")
```

Comment:
I feel like it's more conventional to have group and order after the where clause.

So SELECT .... JOIN  ... WHERE... GROUP ... ORDER ... LIMIT